### DISCUSSION TITLE
chain consult-find and consult-ripgrep

### DISCUSSION THREAD

**Discussion by iris-garcia** [minad/consult#897] [Q&A] [answered]
I know this is a bit off-topic, but is there a way to chain the results given by `consult-find` to `consult-ripgrep`?

Sometimes I'm in situations where this would help a lot.

Thanks in advance.

**Comment by oantolin** [comment, marked as answer]
Well, it doesn't currently work, but once I sort out oantolin/embark#677, you will be able to use `embark-act-all` to call `consult-ripgrep` right from a `consult-find` session. (I have a branch of embark with that functionality but haven't merged it because I needed to carefully think through what around-hooks mean for multitarget action and then I got caught up with end of semester business.)

**Reply by minad** [reply]
Thanks! I am looking forward to that feature.

**Reply by iris-garcia** [reply]
Thank you for the answer, I have been looking into https://github.com/oantolin/embark/issues/677 and even tested it.
However, I find that flow not very convinient.
My ideal solution would be to give the list of files filtered by `consult-find` as glob filter in the `consult-ripgrep` call, this way we just call `rg` once.

**Reply by minad** [reply]
> My ideal solution would be to give the list of files filtered by consult-find as glob filter in the consult-ripgrep call, this way we just call rg once.

I don't have plans to add more functionality for chaining find and grep, but feel free to make a concrete proposal.

**Reply by oantolin** [reply]
Oh, if you just want to pass a glob to consult-grep.command you can do that already. (I forget the exact syntax, but it's something like adding `--` and then any extra parameters you want to give to the executable.) I thought you needed the full power of consult-find, with its second round of filtering inside Emacs.

**Reply by iris-garcia** [reply]
Probably I'm not explaining myself, I'll try to show a common use case I have.
Imagine a folder structure like:
```
.
├── dev
│   ├── file1.yaml
│   ├── file2.yaml
│   └── file3.yaml
├── prod
│   ├── file1.yaml
│   ├── file2.yaml
│   └── file3.yaml
└── staging
    ├── file1.yaml
    ├── file2.yaml
    └── file3.yaml
```
Then all the yaml files have a line like: `version: 1.2.0` and I would like to update the version in all files but only for `staging` and `dev` for example.

I can use `consult-find` to filter the files in which I want to grep later:
![image](https://github.com/minad/consult/assets/3024945/12341632-8c66-46c6-9a0a-933214ec226d)

Then I would use `consult-ripgrep` to search for `version:`, export the matches and edit all files at once.

What I had in mind, is to append the glob filter to `rg` like this:
![image](https://github.com/minad/consult/assets/3024945/c872e90b-9d83-42e7-9881-5633dbafc19b)

**Reply by oantolin** [reply]
OK. Then I don't know why the Embark functionality is not what you want.

**Reply by iris-garcia** [reply]
Probably it is, I just couldn't get it to do that. Is this available by default?
When I try the embark-act-all it will pop the consult-grep command for each of the matches in consult-find.
And using embark-act works only in the file selected.

**Reply by oantolin** [reply]
It is not *yet* available by default (as I said above). As a quick workaround for now try evaluating `(cl-pushnew 'consult-ripgrep embark-multitarget-actions)` (as is mentioned in the issue I linked to) and then try embark-act-all again. It will eventually be even slightly nicer than with that setting.

**Reply by iris-garcia** [reply]
First of all, thank you for your time @oantolin, I already tried that after reading the issue you linked.
But the behavior I get is that `embark-act-all` is running `consult-ripgrep` for each of the files matched with `consult-find`.
This makes impossible to export the results of ripgrep or grep in all of the files filtered by `consult-find`, not sure if you get my point

**Reply by iris-garcia** [reply]
This is what I get:
![output-2023-12-01-16:04:48](https://github.com/minad/consult/assets/3024945/0f5d45fa-a3d7-4bd0-921a-2ec6e346f76b)

**Reply by oantolin** [reply]
That is odd, adding `consult-ripgrep` to the list `embark-multitarget-actions` is meant to deal with exactly the problem you describe and should cause embark-act-all to pass all files in a list argument to `consult-ripgrep` instead of looping over the files and calling `consult-ripgrep` for each one. The day I wrote that comment I did test to see if it worked, so I'm not sure why it is not working for you now.

**Reply by iris-garcia** [reply]
Ok, then probably it is something in my config, maybe the versions I have.
I'll investigate that, thank you @oantolin.

**Reply by iris-garcia** [reply]
I'm very sorry for wasting your time here guys, I hope this helps someone else in the future.
Using the latest version of `embark` did the trick, everything is working as @oantolin described, this improves my workflow a lot. Thank you!

**Reply by oantolin** [reply]
There's no need to apologize, and I'm glad things are working now, @iris-garcia!

**Reply by oantolin** [reply]
With the current commits of embark and embark-consult, this now should work out of the box, @iris-garcia.

**Reply by iris-garcia** [reply]
That's amazing, I'll test it right now. Thank you @oantolin!
